code/datums/components/uplink.dm

```diff
@@ -375,7 +375,7 @@
 
 // PDA signal responses
 
-/datum/component/uplink/proc/new_ringtone(datum/source, atom/source, new_ring_text)
+/datum/component/uplink/proc/new_ringtone(atom/source, new_ring_text)
```

Comment:
The number of args here is right, they've just named the second arg wrong.
```suggestion
/datum/component/uplink/proc/new_ringtone(datum/source, mob/living/user, new_ring_text)
```
Which means you'll have to update the uses of `source` in this proc to use `user`.